### PR TITLE
feature: Add amazon location tracker resource

### PR DESCRIPTION
**Pull request by pocket7878**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #19629 

Output from acceptance testing:

```
❯ make testacc TEST=TestAccLocationTracker_ PKG=location

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/location/... -v -count 1 -parallel 20   -timeout 180m
=== RUN   TestAccLocationTracker_basic
=== PAUSE TestAccLocationTracker_basic
=== RUN   TestAccLocationTracker_disappers
=== PAUSE TestAccLocationTracker_disappers
=== RUN   TestAccLocationTracker_Description
=== PAUSE TestAccLocationTracker_Description
=== RUN   TestAccLocationTracker_Tags
=== PAUSE TestAccLocationTracker_Tags
=== CONT  TestAccLocationTracker_basic
=== CONT  TestAccLocationTracker_Description
=== CONT  TestAccLocationTracker_Tags
=== CONT  TestAccLocationTracker_disappers
--- PASS: TestAccLocationTracker_disappers (39.80s)
--- PASS: TestAccLocationTracker_basic (43.26s)
--- PASS: TestAccLocationTracker_Description (56.82s)
--- PASS: TestAccLocationTracker_Tags (88.33s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/location	89.948s
```

### CODE REVIEW

File: aws/resource_aws_location_tracker.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+		Delete: resourceAwsLocationTrackerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
```
Comment (DrFaust92) [inline aws/resource_aws_location_tracker.go:27]: add validation for len and regex as mentioned at https://docs.aws.amazon.com/location-trackers/latest/APIReference/API_CreateTracker.html#API_CreateTracker_RequestSyntax

File: aws/resource_aws_location_tracker.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+			},
+			"description": {
```
Comment (DrFaust92) [inline aws/resource_aws_location_tracker.go:29]: same as above

File: aws/resource_aws_location_tracker.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+	if v, ok := d.GetOk("kms_key_id"); ok {
+		result.KmsKeyId = aws.String(v.(string))
+	}
+
+	pricingPlan := d.Get("pricing_plan").(string)
+	result.PricingPlan = aws.String(pricingPlan)
+
+	pricingPlanDataSource, ok := d.GetOk("pricing_plan_data_source")
+	if (pricingPlan == "MobileAssetTracking" || pricingPlan == "MobileAssetManagement") && !ok {
```
Comment (DrFaust92) [inline aws/resource_aws_location_tracker.go:151]: use sdk enums

File: aws/resource_aws_location_tracker.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+			"pricing_plan_data_source": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Esri",
+					"Here",
+				}, false),
+			},
+			"tags": tagsSchemaForceNew(),
```
Comment (DrFaust92) [inline aws/resource_aws_location_tracker.go:56]: not sure this should be force new, see [docs](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/contribution-checklists.md#resource-tagging-code-implementation) for updating tags in update func.
More specifically follow the below example:
```
if d.HasChange("tags_all") {
  o, n := d.GetChange("tags_all")
  if err := keyvaluetags.EksUpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
    return fmt.Errorf("error updating tags: %s", err)
  }
}
```

**Review by DrFaust92 — CHANGES_REQUESTED** [review]

File: aws/resource_aws_location_tracker.go (hunk 4 above)
Comment (pocket7878) [reply aws/resource_aws_location_tracker.go:62]: Currently Amazon Location Service does not provider API for modify tracker attributes:
https://docs.aws.amazon.com/location-trackers/latest/APIReference/API_CreateTracker.html

We'll need to add this functionality when the API becomes available.

Comment (DrFaust92) [reply aws/resource_aws_location_tracker.go:62]: resources don't usually support updating tags in their Update action. Location does have a tag resource api available: 
https://docs.aws.amazon.com/cli/latest/reference/location/tag-resource.html

**Comment by pocket7878** [comment]
@DrFaust92 Thank you for your review! 
Implemented tag update & update documentations.

**Comment by pocket7878** [comment]
Sorry for my late reaction. I'v updated my code to resolve conflicts.

**Comment by ewbankkit** [comment]
Superseded by #25466.

**Pull request closed without merging**